main/parse.c

```diff
@@ -1772,6 +1775,26 @@ extern void makeFileTag (const char *const fileName)
 	}
 }
 
+static void makeEOFTag (const char *const fileName)
```

Comment:
You already have the PR opened! :)
Just drop your "refactor" commit and cherry-pick my extra commit. Everything should be ok after that, or so I hope.